src/views/GoTV/StreamManager.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C)  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { get } from "requests";
+import { Stream } from "GoTV";
+import * as preferences from "preferences";
+import { EventEmitter } from "events";
+import { socket } from "sockets";
+
+class StreamManager extends EventEmitter {
```

Comment:
We can make EventEmitter type safe like this:

```
interface Events {
     update: (streams: Stream[]) => void;
}

class StreamManager extends EventEmitter<Events> {
```

That way, if you're using it and you do something like `stream_manager.on("uplate", () => {})` it'll complain at you and let you know you meant `"update"` :)

Reply:
I tried that and Typescript yelled at me:

```
Type 'StreamEvents' does not satisfy the constraint 'EventMap<StreamEvents>'.
  Type 'StreamEvents' is not assignable to type 'Record<"update", any[]>'.
    Types of property 'update' are incompatible.
      Type '(streams: Stream[]) => void' is not assignable to type 'any[]'.ts(2344)
```

Reply:
Ah, looks like I need `TypedEventEmitter<>` 👌🏻👌🏻

Reply:
Pure `EventEmitter<Events>` is preferred over `TypedEventEmitter`, typed event emitter was our own creation prior to the eventemitter3 library adding types, so now new code should just use that, and over time we'll refactor and remove the old `TypedEventEmitter` usage